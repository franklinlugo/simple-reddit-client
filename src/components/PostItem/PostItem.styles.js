import styled from 'styled-components';
import { motion } from 'framer-motion';

export const Container = styled(motion.div)`
  width: 100%;
  background-color: ${(props) => props.theme.darkgray};
  color: ${(props) => props.theme.white};
  padding: 1rem;
  border-bottom: 1px solid ${(props) => props.theme.red};
`;

export const AutorContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, min-content);
  align-items: center;
  column-gap: 1rem;
`;

export const Author = styled.h3`
  margin: 0;
  padding: 0;
  line-height: 1;
  color: ${(props) => props.theme.white};
  white-space: nowrap;
`;

export const Created = styled.h5`
  margin: 1rem 0;
  color: ${(props) => props.theme.white}60;
  white-space: nowrap;
`;

export const ImageContainer = styled.div`
  display: grid;
  grid-template-columns: 100px 1fr;
  align-items: center;
  column-gap: 1rem;
  margin-top: 1rem;
`;

export const Image = styled.img`
  width: 100px;
  height: 100px;
`;

export const Title = styled.p`
  margin: 0;
  padding: 0;
  line-height: 1;
  color: ${(props) => props.theme.white};
`;

export const DismissContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, min-content);
  justify-content: space-between;
  align-items: center;
  margin-top: 1rem;
`;

export const DismissButton = styled.button`
  display: grid;
  grid-template-columns: repeat(2, min-content);
  white-space: nowrap;
  color: ${(props) => props.theme.red};
  border: 1px solid ${(props) => props.theme.red};
  cursor: pointer;
`;

export const Comments = styled.span`
  white-space: nowrap;
`;
