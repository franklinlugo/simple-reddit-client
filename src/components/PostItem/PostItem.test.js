import React from 'react';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PostItem from './PostItem';
import { postsList } from '../../utils/tests/fixtures';

test('PostItem should render', () => {
  const dataMock = { ...postsList[0] };
  const onDismissPostMock = jest.fn();
  const onSetActivePostMock = jest.fn();
  const { asFragment } = render(
    <PostItem data={dataMock} onDismissPost={onDismissPostMock} onSetActivePost={onSetActivePostMock} />,
  );

  expect(asFragment()).toMatchSnapshot();

  userEvent.click(screen.getByText('111 author'));
  expect(onSetActivePostMock).toBeCalledTimes(1);

  userEvent.click(screen.getByRole('button'));
  expect(onDismissPostMock).toBeCalledTimes(1);
});
