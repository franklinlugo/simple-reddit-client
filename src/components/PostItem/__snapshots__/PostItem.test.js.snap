// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostItem should render 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  padding: 1rem;
  border-bottom: 1px solid;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
}

.c2 {
  margin: 0;
  padding: 0;
  line-height: 1;
  white-space: nowrap;
}

.c3 {
  margin: 1rem 0;
  color: 60;
  white-space: nowrap;
}

.c4 {
  display: grid;
  grid-template-columns: 100px 1fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  margin-top: 1rem;
}

.c5 {
  width: 100px;
  height: 100px;
}

.c6 {
  margin: 0;
  padding: 0;
  line-height: 1;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1rem;
}

.c8 {
  display: grid;
  grid-template-columns: repeat(2,min-content);
  white-space: nowrap;
  border: 1px solid;
  cursor: pointer;
}

.c9 {
  white-space: nowrap;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        111 author
      </h3>
      <h5
        class="c3"
      >
        5/5/1970, 9:00:00 PM
      </h5>
    </div>
    <div
      class="c4"
    >
      <img
        class="c5"
        src="https://image1111.com"
      />
      <p
        class="c6"
      >
        111 title
      </p>
    </div>
    <div
      class="c7"
    >
      <button
        class="c8"
      >
        Dismiss Post
      </button>
      <span
        class="c9"
      >
         0 Comments
      </span>
    </div>
  </div>
</DocumentFragment>
`;
